### PR TITLE
usermod: Small optimization using memmove for password unlock

### PR DESCRIPTION
**Pull request by nertpinx**
And one extra optimization using memmove instead of manually moving bytes.

### CODE REVIEW

**Comment by ikerexxe** [comment]
@hallyn can you take a look at this PR?

File: src/usermod.c [hunk 1]
```diff
@@ -443,11 +441,7 @@ static char *new_pw_passwd (char *pw_pass)
 		              "updating password", user_newname, user_newid, 0);
 #endif
 		SYSLOG ((LOG_INFO, "unlock user '%s' password", user_newname));
-		s = pw_pass;
-		while ('\0' != *s) {
-			*s = *(s + 1);
-			s++;
-		}
+		memmove(pw_pass, pw_pass + 1, strlen(pw_pass) + 1);
```
Comment (hallyn) [inline src/usermod.c:444]: hm, pw_pass+1 is not right for the length, is it?  You're starting at pw_pass + 1, so you'll be accessing one byte past the end of the \0 ending pw_pass.

File: src/usermod.c [hunk 2]
```diff
@@ -443,6 +443,12 @@ static char *new_pw_passwd (char *pw_pass)
 		SYSLOG ((LOG_INFO, "unlock user '%s' password", user_newname));
 		memmove(pw_pass, pw_pass + 1, strlen(pw_pass) + 1);
 	} else if (pflg) {
+		if (strchr(user_pass, ':') != NULL) {
```
Comment (hallyn) [inline src/usermod.c:446]: Should valid_field() be called on user_pass?

File: src/usermod.c (hunk 1 above)
Comment (nertpinx) [reply src/usermod.c:444]: You're right, due to omission of temporary variable I was counting the wrong length, thanks for noticing.

File: src/usermod.c (hunk 2 above)
Comment (nertpinx) [reply src/usermod.c:446]: It actually is, in `lib/pwio.c` which is used to update the password entry anyway. I guess it was an issue elsewhere and I cannot reproduce it now.  I probably just completely missed that.

Comment (nertpinx) [reply src/usermod.c:446]: I-m dropping this patch then and changing the PR name to reflect the only change left in here, sorry for the noise.

**Comment by ikerexxe** [comment]
@hallyn can you merge this PR if you agree with it?

**Review by hallyn — APPROVED** [review]

**Pull request merged**